styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -265,16 +286,31 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, Instant nextNaturalT
       final Entity entity = result.next();
       Workflow workflow;
       try {
-        workflow =
-            Json.OBJECT_MAPPER.readValue(entity.getString(PROPERTY_WORKFLOW_JSON), Workflow.class);
+        workflow = OBJECT_MAPPER.readValue(entity.getString(PROPERTY_WORKFLOW_JSON), Workflow.class);
       } catch (IOException e) {
         LOG.warn("Failed to read workflow {}.", entity.key());
         continue;
       }
-      map.put(workflow,
-          entity.contains(PROPERTY_NEXT_NATURAL_TRIGGER)
-          ? Optional.of(datetimeToInstant(entity.getDateTime(PROPERTY_NEXT_NATURAL_TRIGGER)))
-          : Optional.empty());
+
+      if (entity.contains(PROPERTY_NEXT_NATURAL_TRIGGER)) {
+        Instant instant = datetimeToInstant(entity.getDateTime(PROPERTY_NEXT_NATURAL_TRIGGER));
+        Instant triggerInstant;
+
+        // todo: this check is only needed during a transition period
```

Comment:
Would you like to perform a data migration by adding a new property with default value `PT0S`? Or we do it later after GA.